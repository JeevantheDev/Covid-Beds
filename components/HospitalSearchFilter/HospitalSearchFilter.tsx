/* eslint-disable react/prop-types */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import { TextField, Button } from '@material-ui/core';

export interface Iprops {
  onSearchSubmit: (state: string, zipcode: string) => void;
}

export const HospitalSearchFilter: React.FC<Iprops> = ({ onSearchSubmit }) => {
  const [locationInput, setLocationInput] = useState<{ state: string; zipcode: string }>({
    state: '',
    zipcode: '',
  });

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setLocationInput({
      ...locationInput,
      [e.target.name]: e.target.value,
    });
  };

  const handleSearchSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    onSearchSubmit(locationInput.state, locationInput.zipcode);
  };

  const { state, zipcode } = locationInput;

  return (
    <form data-testid="search-section-form">
      <Grid container spacing={2}>
        <Grid item xs={6} md={6} sm={6}>
          <TextField
            inputProps={{ 'data-testid': 'state' }}
            value={state}
            onChange={handleInputChange}
            fullWidth
            variant="outlined"
            placeholder="State"
            name="state"
          />
        </Grid>
        <Grid item xs={6} md={6} sm={6}>
          <TextField
            inputProps={{ 'data-testid': 'zipcode' }}
            value={zipcode}
            onChange={handleInputChange}
            fullWidth
            variant="outlined"
            placeholder="ZipCode"
            name="zipcode"
          />
        </Grid>
        <Grid item xs={12} md={12} sm={12}>
          <Button
            data-testid="submit"
            onClick={handleSearchSubmit}
            style={{ width: '100%' }}
            color="secondary"
            variant="contained"
          >
            Find Hospital
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};
