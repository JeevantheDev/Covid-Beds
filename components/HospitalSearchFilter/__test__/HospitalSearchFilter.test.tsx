/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { HospitalSearchFilter, Iprops } from '../HospitalSearchFilter';
import '@testing-library/jest-dom/extend-expect';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

function renderFilterSearchForm(props: Partial<Iprops> = {}) {
  const defaultProps: Iprops = {
    onSearchSubmit(state: string, zipcode: string) {
      return;
    },
  };

  return render(<HospitalSearchFilter {...defaultProps} {...props} />);
}

test('should display a blank search box', async () => {
  const { findByTestId } = renderFilterSearchForm();
  const searchSectionForm = await findByTestId('search-section-form');
  expect(searchSectionForm).toHaveFormValues({
    state: '',
    zipcode: '',
  });
});

test('should allow user to submit for search', async () => {
  const onSearchSubmit = jest.fn();
  const { findByTestId } = renderFilterSearchForm({ onSearchSubmit });
  const state = await findByTestId('state');
  const zipcode = await findByTestId('zipcode');
  const submit = await findByTestId('submit');

  fireEvent.change(state, { target: { value: 'Odisha' } });
  fireEvent.change(zipcode, { target: { value: '751003' } });
  fireEvent.click(submit);

  expect(onSearchSubmit).toHaveBeenCalledWith('Odisha', '751003');
});

it('matches snapshot', () => {
  const onSearchSubmit = jest.fn();
  const tree = TestRenderer.create(
    <HospitalSearchFilter onSearchSubmit={onSearchSubmit}></HospitalSearchFilter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
