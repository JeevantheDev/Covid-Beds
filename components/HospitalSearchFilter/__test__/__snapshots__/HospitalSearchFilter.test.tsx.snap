// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  data-testid="search-section-form"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="state"
            disabled={false}
            name="state"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="State"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-10"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="zipcode"
            disabled={false}
            name="zipcode"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="ZipCode"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-10"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        data-testid="submit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Find Hospital
        </span>
      </button>
    </div>
  </div>
</form>
`;
